src/throttledPromiseAll.ts

```diff
@@ -124,14 +134,43 @@ export class ThrottledPromiseAll<T, O = T> {
   }
 
   private async dequeue(): Promise<void> {
-    while (this.queue.length > 0) {
-      const next = this.queue.slice(0, this.concurrency);
-      this.queue.splice(0, this.concurrency);
+    const generator = function* (
+      data: Array<PromiseItem<T, O | undefined>>
+    ): Generator<PromiseItem<T, O | undefined> | undefined> {
+      while (data.length > 0) {
+        yield data.shift();
+      }
+    };
+    const concurrencyPool: Map<number, Promise<IndexedResult<O> | undefined>> = new Map<
```

Comment:
Make pool a map, which simplifies the removal of pool entries that have resolved.